atlas/schema.hcl

```diff
@@ -1,11 +1,476 @@
-table "test" {
+table "expense_snapshot_v1" {
   schema = schema.public
-  column "id" {
+  column "expense_id" {
     null = false
-    type = integer
+    type = bigint
+  }
+  column "snapshot_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "scan_id" {
+    null = true
+    type = bigint
+  }
+  column "title" {
+    null = false
+    type = text
+    comment = "name of item/service expense"
+  }
+  column "amount" {
+    null = false
+    type = money
+  }
+  column "deductible" {
+    null = false
+    type = double_precision
+    comment = "percentage of expense that is deductible"
+  }
+  column "create_time" {
+    null = false
+    type = timestamptz
+  }
+  primary_key {
+    columns = [column.expense_id, column.snapshot_id]
+  }
+  foreign_key "expense_id" {
+    columns     = [column.expense_id]
+    ref_columns = [table.expense_v1.column.expense_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+  foreign_key "scan_id" {
+    columns     = [column.scan_id]
+    ref_columns = [table.scan_v1.column.scan_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+}
+table "expense_v1" {
+  schema = schema.public
+  column "project_id" {
+    null = false
+    type = bigint
+  }
+  column "expense_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  primary_key {
+    columns = [column.project_id, column.expense_id]
+  }
+  foreign_key "project_id" {
+    columns     = [column.project_id]
+    ref_columns = [table.project_v1.column.project_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+index "expense_id_idx" {
+    unique  = true
+    columns = [column.expense_id]
+  }
+}
+table "income_snapshot_v1" {
+  schema = schema.public
+  column "income_id" {
+    null = false
+    type = bigint
+  }
+  column "snapshot_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "scan_id" {
+    null = true
+    type = bigint
+  }
+  column "title" {
+    null = false
+    type = text
+    comment = "name of product/service source of income"
```

Comment:
please use full sentences ending with a comma here and other places

Reply:
a full sentence that ends with a comma not a period?

Reply:
sorry, period lol